duckduckgo/iOS
# Add NetP widget

Task/Issue URL: https://app.asana.com/0/414235014887631/1205921206830836/f
Tech Design URL:
CC: @graeme 

**Description**:

This PR adds a NetP widget. It's only available in the debug and alpha builds, and only available on iOS 17.



**Steps to test this PR**:
1. First, update the `#if ALPHA` checks in this PR to include `DEBUG` - the alpha checks are because we would like to get the widget into the alpha builds, but we need CI to support Xcode 15, which we can't do for any workflows that run tests unfortunately
1. Run the app on your device
2. Add the widget from the widget UI
3. Toggle NetP within the app, check that the widget state updates accordingly
4. Try the "Connect" and "Disconnect" buttons inside the widget - note that the widget does not show connecting/disconnecting states, this functionality is pretty hacky at the moment



**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
